torch/csrc/jit/codegen/cuda/kernel_expr_evaluator.h

```diff
@@ -37,13 +43,19 @@ class TORCH_CUDA_CU_API ExpressionEvaluator : private IrVisitor {
 
   //! Try to evaluate a Kernel IR value
   c10::optional<Int::ScalarType> evaluate(const Val* value);
+  c10::optional<Int::ScalarType> evaluateMemoized(const glfdc::ExprEvaluator&);
```

Comment:
Why don't we simply call this `evaluate`?

Reply:
Renamed